slack_sdk/models/blocks/basic_components.py

```diff
@@ -559,3 +559,25 @@ def to_dict(self) -> Dict[str, Any]:  # skipcq: PYL-W0221
         else:
             json["trigger"] = self._trigger
         return json
+
+
+class SlackFile(JsonObject):
+    attributes = {"id", "url"}
+
+    def __init__(
+        self,
+        *,
+        id: Optional[str] = None,
+        url: Optional[str] = None,
```

Comment:
There's no way to express an "either/or" constraint in python types, right?

Reply:
yeah, it could be complicated like having two classes and forcing devs to instantiate them instead